src/mlpack/core/data/split_data.hpp

```diff
@@ -61,9 +64,17 @@ void Split(const arma::Mat<T>& input,
   trainLabel.set_size(trainSize);
   testLabel.set_size(testSize);
 
-  const arma::Col<size_t> order =
-      arma::shuffle(arma::linspace<arma::Col<size_t>>(0, input.n_cols - 1,
-                                                      input.n_cols));
+  arma::Col<size_t> order;
+  if (shuffleData)
+  {
+    order = arma::shuffle(arma::linspace<arma::Col<size_t>>(0,
+        input.n_cols - 1, input.n_cols));
+  }
+  else
+  {
+    order = arma::linspace<arma::Col<size_t>>(0, input.n_cols - 1,
+        input.n_cols);
+  }
```

Comment:
Just one comment from me---would you mind redoing the implementation so that `order` is not used at all if `shuffleData` is `false`?  This implementation will unnecessarily create a (potentially big) vector `order` and do a bunch of unnecessary work, when we could simplify and just use a direct call to `input.cols()` and `inputLabel.subvec()`.

Reply:
Done :) ✅